### ISSUE TITLE
Ideas for developing the Stargate Network mod.

### ISSUE THREAD

**Issue by ilaore** [AlmuraDev/SGCraft#170]
I suggest adding the following to the Stargate Network mod:

1. a New dimension (Abydos. The same planet near which RA died, in the 1994 film Stargate). The only way to go to Abydos was through the Stargate. And the address of the gate of Abydos, which we need to go to it, I suggest adding to the danzhi that will appear in the normal world. And that in each new world, the address of the Stargate of Abydos was different.
2. This is already a little easier to do, but it greatly diversifies the game with the Stargate Network mod. I propose to do so through our Stargate, and sometimes came some new mobs, such as the same Jaffa. Of course, the Jaffa are supposed to be aggressive and attack us.

**Comment by Bysokar** [comment]
@ilaore Use CustomNPCs to generate your Jaffa and Use AdvancedRocketry to make randomised dimensions.

They should be separate from SGCraft itself.

(I run StargateMC.com, and that's how we did it)

**Comment by gaiaweylyn** [comment]
If you’re not already familiar with it, I strongly recommend checking out ATUM 2, which adds in an incredible desert dimension, complete with pyramids and ruins as part of the worldgen. The sandstorms in there are pretty intense too. 

It’s what I personally use to represent Abydos anyways, though do keep in mind that it can get pretty tough there on survival if you’re not prepared, especially on normal and hard difficulties. If it becomes a problem, editing ATUM’s config might help.

**Comment by gaiaweylyn** [comment]
As for creating completely custom dimensions, I’d recommend looking into Just Enough Dimensions. Its an incredibly powerful tool with a steep learning curve, but it’s absolutely worth it. I use it in combination with Biomes o Plenty to generate dimensions filled only using specified biomes. By combining biomes of a particular color scheme for the dimension in mind, this enables the easy creation of many varied worlds, each one visually distinct to the next, not to mention the overworld.

On top of that, each dimension can be assigned a custom day/night cycle. Perhaps your world is meant to have a slower rotation, so you just make it last longer. Its even possible to disable cycle progression while frozen at a specific time, so you can have perpetual night or dusk, perhaps you’re on a tidally locked planet.

JED does not provide any tangible means to access the new dimensions, so instead it adds an upgraded teleport command that can be used between any dimension. Use the command for the first visit to each dimension. Set up a Stargate and fueled DHD in each one, save the addresses, and you’re all set. If you’re curious about all the other stuff JED can do, just check out the manual.

**Comment by Bysokar** [comment]
@gaiaweylyn JED is great, AdvancedRocketry is better though :)  It ties it all dimensions into a single universe and adds realism and spacecraft to travel between stars when you blow up your gate.

It also requires little to no configuration, it has a universe generator built-in.

**Comment by gaiaweylyn** [comment]
Ah, fair point. I guess I like having complete control over every option and variable when using JED. When making a multi dimension Stargate adventure map for instance, I want everything to be *just* right.

So I could definitely see both options being ideal for different players or scenarios. I also vaguely remember running into some kind of mod incompatibility with Advanced Rocketry, I just don’t remember with what. I’ll probably mess with both and see if I chance across it.

**Comment by Bysokar** [comment]
Check out stargatemc.com if you'd like a good example of AR done well :D @gaiaweylyn

**Comment by gaiaweylyn** [comment]
@draksterau I just might. On a related note to the OP’s second question. Have you guys ever come up with good methods for creating the illusion of hostile mob/npc sieges through an open gate? Well, methods other than the below one I came up with anyways..

My idea was to put a caged villager behind a stargate, dial, activate a redstone ticker dispenser filled with zombie eggs and quickly retreat through the gate. The active stargate keeps the chunk loaded after I leave, and the lure draws them straight through the gate, where they retarget their AI and attack.

And it worked, a steady stream of zombies followed right behind me, and continued coming until the dispenser ran out and the gate timed out. Unfortunately, this can only work with mobs that can be drawn to a non-player entity, and I never got any further with it.

**Comment by Bysokar** [comment]
@gaiaweylyn Yes, you need to write a mod that can generate customnpc clones at a position on demand & activates gates.
We have that functionality just about ready to go.. Ive done it in the past.
